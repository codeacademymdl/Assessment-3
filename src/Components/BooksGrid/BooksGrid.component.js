import React, { Component } from 'react';
// import './BooksGrid.css';
import Books from '../Books/Books.component'

class BooksGrid extends Component {
    render() {
        // console.log(this.props)
        return (
            
            
            <div>
                <fieldset>
                    <legend align="center">{this.props.author}</legend>
                    {
                        this.props.books.map((book,index)=>{
                            return <Books
                            Name={book.Name}
                            rating={book.rating}
                            imageSrc = {`./cover_image.jpeg`}
                            
                            ></Books>
                        })
                    }

                </fieldset>
                </div>

            
        )
    }
}

export default BooksGrid;
