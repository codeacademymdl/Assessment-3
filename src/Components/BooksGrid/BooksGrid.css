.authorGrid{

  width:80%;
  border-width:10px;  
border-style:solid;
border-color: #FC9152;
border-radius: 40px;
font-size: 2rem;
margin-top:5%;
min-width: 20%;

}

.authorGrid legend{
    background-color: white;
    border-radius: 5px;
    color:#FC9152;
    font-size:1em;
    padding:0.2em 0.5em;
    /* width:20%; */
   
   
}

.main{
display:flex;   
/* align-content: center; */
flex-direction: row;
justify-content: center;

}

.innerDiv{
    display:flex;
    flex-wrap: wrap;      
   
    
}