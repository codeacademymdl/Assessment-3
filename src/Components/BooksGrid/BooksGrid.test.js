import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import BooksGrid from './BooksGrid.component'

describe('BooksGrid', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<BooksGrid author="J K Rowling"
           key="1" 
           book={
               [{
            "Name": "Harry Potter and the Sorcerers Stone (Harry Potter, #1)",
            "rating": "4.45",
            "imageSrc": "{`./cover_image.jpeg`}"
        }]
    }/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
});

