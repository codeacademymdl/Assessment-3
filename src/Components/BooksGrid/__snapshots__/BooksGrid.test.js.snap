// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksGrid renders correctly 1`] = `
<div
  className="main"
>
  <fieldset
    className="authorGrid"
  >
    <legend
      align="center"
    >
      J K Rowling
    </legend>
    <div
      className="book"
    >
      <img
        alt="book"
        className="image"
        src="./cover_image.jpeg"
      />
      <div
        className="container"
      >
        <p
          className="title"
        >
          Harry Potter and the Sorcerers Stone (Harry Potter, #1)
        </p>
        <span>
          4.45
        </span>
        <img
          alt=""
          className="heart"
          onClick={[Function]}
          src="heart-black.svg"
        />
      </div>
    </div>
  </fieldset>
</div>
`;
